### PR TITLE
set core.stdc.stdint types so mangling matches C++

### PR DESCRIPTION
**Pull request by WalterBright**
Provides better interoperability with C++, hence betterC.

### CODE REVIEW

File: src/core/stdc/config.d [hunk 1]
```diff
@@ -15,17 +15,35 @@ module core.stdc.config;
...
 {
+    private
+    {
+        version (Posix)
+            enum isPosix = true;
+        else
+            enum isPosix = false;
+        static if (isPosix && (void*).sizeof > int.sizeof)
+        {
+            alias ddoc_long = long;
```
Comment (JinShil) [inline src/core/stdc/config.d]: What the heck is a `ddoc_` type?

Comment (JinShil) [inline src/core/stdc/config.d]: I don't see why we need these itermediate `ddoc_` aliases here.  Can't we just declare `c_long` and `c_ulong` here.

**Comment by JinShil** [comment]
There's an awful lot of redundancy in this patch and it makes review a little difficult.  Can't we just version out the differences?

**Comment by WalterBright** [comment]
> There's an awful lot of redundancy in this patch and it makes review a little difficult. Can't we just version out the differences?

I tried that initially. The result was incomprehensible.

**Comment by WalterBright** [comment]
I put you on the team. Druntime can use your help.

**Pull request merged**